mypy/semanal.py

```diff
@@ -3508,7 +3510,9 @@ def lookup_qualified(self, name: str, ctx: Context,
                         break
                 if n:
                     n = self.normalize_type_alias(n, ctx)
-            return n
+                    if n and n.module_hidden:
+                        self.name_not_defined(name, ctx)
+            return n if n and not n.module_hidden else None
```

Comment:
Minor style nit. Take it if you want. I prefer to not have compound if statements in returns. It makes things clearer if you break it up, but up to you.